packages/cli/generation/ir-generator/src/examples/generator/generatePrimitiveExample.ts

```diff
@@ -16,8 +23,12 @@ export function generatePrimitiveExample({
 }: generatePrimitiveExample.Args): ExampleGenerationSuccess<ExamplePrimitive> {
     switch (primitiveType.v1) {
         case "STRING": {
-            const jsonExample = fieldName ?? Examples.STRING;
-            return { type: "success", example: ExamplePrimitive.string({ original: jsonExample }), jsonExample };
+            let validation = undefined;
+            if (primitiveType.v2?.type === "string") {
+                const stringType = primitiveType.v2 as PrimitiveTypeV2.String;
+                validation = stringType.validation;
+            }
```

Comment:
Can we roll out a few helper functions for this? Something along the lines of `maybeDoubleValidation`, `maybeStringValidation`, and so on? That way we can get rid of the dangling `let` and just call it in-line (since `undefined` is a valid value. 

I'd also make it so that the new function below takes in named arguments so the call-site looks like the following:

```typescript
return generatePrimitiveStringExample({
    fieldName,
    validation: maybeDoubleValidation(primitiveType.v2)
})
```

I'll leave a comment below to demonstrate how that's done.

Reply:
implemented